File: Python_Bluetooth_Arduino/DataConverter/DataConverter.py
def DataConverter(dataArr, expansion):
        res = [0]*len(dataArr)*expansion
        for i in range(len(dataArr)):
            sign = 1
            current_val = dataArr[i]
            if current_val < 0:
                sign = -1
                current_val = -current_val
            for j in range(expansion):
                res[i*expansion+expansion-j-1] = sign*(current_val%100)
                current_val = (current_val - (current_val%100))/100
        return res    

print(DataConverter([5000, 1000, 2000, 4899, -1200, 3200], 3))

File: Python_Bluetooth_Arduino/DataConverter/DataConverter.cpp
#include <iostream>

using namespace std;

int main(){
    int expansion = 3;
    int num = 18;
    int data[num] = {0, 50, 0, 0, 10, 0, 0, 20, 0, 0, 48, 99, 0, -12, 0, 0, 32, 0};
    int data_final[num/expansion];
    for (int i = 0; i < num; i++){
      int sign = 1;
      int current_res = 0;
      for (int j = 0; j < expansion; j++){
        int current_val = data[i*expansion+j];
        if (current_val < 0)
        {
            sign = -1;
            current_val = -current_val;
        }
        current_res = current_res*100;
        current_res += current_val;
      }
      data_final[i] = sign*current_res;
    }
    for (int i = 0; i < num/expansion; i++)
    {
        cout << data_final[i] << " ";
    }
    cout << endl;
    
}